contracts/cw20-base/helpers.ts

```diff
@@ -265,6 +268,16 @@ export const CW20 = (client: SigningCosmWasmClient, fees: Options["fees"]): CW20
       return result.transactionHash;
     }
 
+    const send = async (senderAddress: string, recipient: string, amount: string, msg: Record<string, unknown>): Promise<string> => {
+      const result = await client.execute(senderAddress, contractAddress, {send: {recipient, amount, msg: toUtf8(JSON.stringify(msg))}}, fees.exec);
```

Comment:
One more step needed here (maybe you can pull this out to a function `jsonToBinary`

`msg: toBase64(toUtf8(JSON.stringify(msg)))` 

We want to explicitly make it a base64-encoded string. I think the default JSON serialisation in JS land for Uint8Array is `[43, 91, 81, ...]`, which is not what we want